### PR TITLE
feat: add closure to `Verifier`

### PR DESCRIPTION
**Pull request by felix-berg**

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/language/phase/Verifier.scala [hunk 1]
```diff
@@ -302,6 +302,21 @@ object Verifier {
             case _ => failMismatchedShape(t1, "Tuple", loc)
           }
 
+        case AtomicOp.Closure(sym) =>
+          val defn = root.defs(sym)
+          if (defn.cparams.length != ts.length)
+            failMismatchedShape(ts, s"List of types matching closure parameters of $sym", loc)
```
Comment (JonathanStarup) [inline main/src/ca/uwaterloo/flix/language/phase/Verifier.scala:308]: This error can be better but I'm not sure exactly how. Right now the error says `expected = 'List of types matching closure parameters of clo$2761', found = List(Int32, Int32)`. it doesn't say what length or types it expected.

I think more work can be put into creating some nice error - but a simpler decent solution could be to take the approach of ApplyDef where a big arrow type is built, the `(ts..) -> tpe` should be the same as `(defn.cparams..) -> (defn.fparams..) -> defn.tpe`.

File: main/src/ca/uwaterloo/flix/language/phase/Verifier.scala [hunk 2]
```diff
@@ -302,6 +302,21 @@ object Verifier {
...
+
+          val ctps = defn.cparams.map(cp => cp.tpe)
```
Comment (JonathanStarup) [inline main/src/ca/uwaterloo/flix/language/phase/Verifier.scala:310]: `_.tpe` is shorthand for `cp => cp.tpe`

File: main/src/ca/uwaterloo/flix/language/phase/Verifier.scala (hunk 1 above)
Comment (felix-berg) [reply main/src/ca/uwaterloo/flix/language/phase/Verifier.scala:308]: yea thats way better

**Review by JonathanStarup — APPROVED** [review]

**Review by magnus-madsen — APPROVED** [review]

**Comment by magnus-madsen** [comment]
Thanks @felix-berg

**Pull request merged**